lib/ClangImporter/SwiftDeclSynthesizer.cpp

```diff
@@ -2538,6 +2538,9 @@ llvm::SmallVector<clang::CXXMethodDecl *, 4>
 SwiftDeclSynthesizer::synthesizeStaticFactoryForCXXForeignRef(
     const clang::CXXRecordDecl *cxxRecordDecl) {
 
+  if (cxxRecordDecl->isAbstract())
```

Comment:
Do we have a test covering this branch? I am ok with the change but would be nice to have it actually tested.